pkg/prometheus/promcfg.go

```diff
@@ -3056,12 +3054,7 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 				})
 			}
 
-			if config.Filters != nil {
-				configs[i] = append(configs[i], yaml.MapItem{
-					Key:   "filters",
-					Value: config.Filters,
-				})
-			}
+			configs[i] = cg.addFiltersToYaml(configs[i], "2.3.0", config.Filters)
```

Comment:
(suggestion) a bit more verbose but more readable from my POV (you don't have to ask yourself what this "2.3.0" string is about). I bet that it's also simpler for Docker SDs since you don't have to pass the version again.
```suggestion
			configs[i] = cg.WithMinimumVersion("2.3.0").addFiltersToYaml(configs[i], config.Filters)
```

Reply:
> I bet that it's also simpler for Docker SDs since you don't have to pass the version again.
Not sure I understand this.